precompiles/batch/src/tests.rs

```diff
@@ -0,0 +1,638 @@
+                .write(vec![Address(Bob.into())])
+                .write(vec![U256::from(1u8)])
+                .write(vec![Bytes::from(&one[..])])
+                .write::<std::vec::Vec<U256>>(vec![])
+                .build(),
+        )
+        .with_target_gas(Some(total_call_cost - 1))
+        .with_subcall_handle(move |_subcall| panic!("there shouldn't be any subcalls"))
+}
+
+#[test]
+fn batch_all_call_out_of_gas() {
+    ExtBuilder::default().build().execute_with(|| {
+        batch_call_out_of_gas(&precompiles(), Mode::BatchAll).execute_error(ExitError::OutOfGas);
+    })
+}
+
+#[test]
+fn batch_some_call_out_of_gas() {
+    ExtBuilder::default().build().execute_with(|| {
+        batch_call_out_of_gas(&precompiles(), Mode::BatchSome)
+            .expect_log(log_subcall_failed(precompile_address(), 0))
+            .execute_returns(EvmDataWriter::new().write(true).build());
+    })
+}
+
+#[test]
+fn batch_some_until_failure_call_out_of_gas() {
+    ExtBuilder::default().build().execute_with(|| {
+        batch_call_out_of_gas(&precompiles(), Mode::BatchSomeUntilFailure)
+            .expect_log(log_subcall_failed(precompile_address(), 0))
+            .execute_returns(EvmDataWriter::new().write(true).build());
+    })
+}
+
+fn batch_gas_limit(
+    precompiles: &BatchPrecompileMock<Runtime>,
+    mode: Mode,
+) -> PrecompilesTester<BatchPrecompileMock<Runtime>> {
+    let (_, total_call_cost) = costs();
+    let one = b"one";
+
+    precompiles
+        .prepare_test(
+            Alice,
+            precompile_address(),
+            EvmDataWriter::new_with_selector(check_mode(mode))
+                .write(vec![Address(Bob.into())])
+                .write(vec![U256::from(1u8)])
+                .write(vec![Bytes::from(&one[..])])
+                .write::<std::vec::Vec<U256>>(vec![U256::from(50_000 - total_call_cost + 1)])
+                .build(),
+        )
+        .with_target_gas(Some(50_000))
+        .with_subcall_handle(move |_subcall| panic!("there shouldn't be any subcalls"))
+}
+
+#[test]
+fn batch_all_gas_limit() {
+    ExtBuilder::default().build().execute_with(|| {
+        batch_gas_limit(&precompiles(), Mode::BatchAll).execute_error(ExitError::OutOfGas);
+    })
+}
+
+#[test]
+fn batch_some_gas_limit() {
+    ExtBuilder::default().build().execute_with(|| {
+        let (return_log_cost, _) = costs();
+
+        batch_gas_limit(&precompiles(), Mode::BatchSome)
+            .expect_log(log_subcall_failed(precompile_address(), 0))
+            .expect_cost(return_log_cost)
+            .execute_returns(EvmDataWriter::new().write(true).build());
+    })
+}
+
+#[test]
+fn batch_some_until_failure_gas_limit() {
+    ExtBuilder::default().build().execute_with(|| {
+        batch_gas_limit(&precompiles(), Mode::BatchSomeUntilFailure)
+            .expect_log(log_subcall_failed(precompile_address(), 0))
+            .execute_returns(EvmDataWriter::new().write(true).build());
+    })
+}
```

Comment:
I wonder why many of the unit tests in the original Moombean implementation were not included here?

Reply:
for the rest of the tests they use macros which we don't have, and we decide to add it, it will be a big edition to the code and we will have to refactor all the precompiles that we have. 
The tests that we have covers all the functions in the precompile.